### ISSUE TITLE
Ruby 3 - Upgrade Management & Camerata

### ISSUE THREAD

**Issue by mikeapp** [yalelibrary/YUL-DC#2512]
**Story**

Move to Ruby 3, perhaps  https://hub.docker.com/r/phusion/passenger-ruby30 or later image?

I'm confused about this earlier PR: https://github.com/yalelibrary/yul-dc-camerata/pull/356 , it seems to leave us on 2.7?

**Acceptance**
- [x] Confirm what version we're currently on
- [x] Confirm what version we can/should move to with Amanda
- [x] Ruby images are using Ruby 3

**Comment by K8Sewell** [comment]
We're on 3.1.3 it seems.  


![Image](https://github.com/yalelibrary/YUL-DC/assets/36549923/65a85add-39d5-46a6-81fb-243da6fdff5a)



However, I drafted a PR to upgrade our phusion passenger image to match and be less confusing.  Will await information from Amanda to finalize the Ruby version we can jump to.  https://github.com/yalelibrary/yul-dc-camerata/pull/365

**Comment by DraxIndustries79** [comment]
Amanda confirms that there are no conflicts between Ruby 3.2.0 and Blacklight 7.33.1, we can target 3.2.0.

**Comment by K8Sewell** [comment]
Working through these errors to fix the failing tests on management



![Image](https://github.com/yalelibrary/YUL-DC/assets/36549923/8ae0478f-041f-4bfe-9294-f3ed39a3b3ca)



![Image](https://github.com/yalelibrary/YUL-DC/assets/36549923/7f66bd41-9866-462d-b061-06be353ba55b)

**Comment by DraxIndustries79** [comment]
Set aside till the multiple jobs issues is resolved

**Comment by K8Sewell** [comment]
## Progress Update on test repair work:

Failing tests after upgrade started near 450 but we're now at 120.


#### Channels - 0

#### Controllers - 0

#### Datatables - 2

- [ ] 	Please stub a default value first if message might be received with other args as well - admin_set_datatable_spec.rb:16
- [ ] 	undefined method `admin_set_path' for nil:NilClass - admin_set_datatable_spec.rb:10
	4
#### Jobs - 30

- [ ] 	5 - Failure/Error: ActiveJob::Scheduler.start (replace with good job scheduler) - /spec/jobs/activity_stream_job_spec.rb:28
- [ ] 	1 - expected `GoodJob::Job.count` to have changed by 1, but was changed by 0 - create_child_oid_csv_job_spec.rb:14
- [ ] 	1 - expected `GoodJob::Job.count` to have changed by 1, but was changed by 0 - /spec/jobs/generate_manifest_job_spec.rb:15
- [ ] 	1 - expected `GoodJob::Job.count` to have changed by 1, but was changed by 0 - /spec/jobs/generate_output_csv_job_spec.rb:13
- [ ] 	3 - wrong number of arguments (given 2, expected 1) - /spec/jobs/generate_pdf_job_spec.rb:41
- [ ] 	3 - expected `GoodJob::Job.count` to have changed by 1, but was changed by 0 - /spec/jobs/generate_ptiff_job_spec.rb:34
- [ ] 	1 - expected `GoodJob::Job.count` to have changed by 1, but was changed by 0 - /spec/jobs/mets_directory_scan_job_spec.rb:12
- [ ] 	1 -  Failure/Error: ActiveJob::Scheduler.start (replace with good job scheduler) - /spec/jobs/problem_report_job_spec.rb:20
- [ ] 	1 - expected `GoodJob::Job.count` to have changed by 1, but was changed by 0 - /spec/jobs/problem_report_manual_job_spec.rb:15
- [ ] 	2 - info instead of failed status & good job count not changed - /spec/jobs/reassociate_child_oids_job_spec.rb:13
- [ ] 	1 - good job count not changed - /spec/jobs/recreate_child_oid_ptiffs_job_spec.rb:44
- [ ] 	2 - good job count not changed & nil class issue - /spec/jobs/setup_metadata_job_spec.rb:13
- [ ] 	4 - good job count not changed - /spec/jobs/solr_index_job_spec.rb:13
- [ ] 	? - stalled - /spec/jobs/update_all_metadata_job_spec.rb
- [ ] 	? - stalled - spec/jobs/update_fulltext_status_job_spec.rb
- [ ] 	1 - good job count not changed - /spec/jobs/update_manifests_job_spec.rb:20
- [ ] 	1 - good job count not changed - /spec/jobs/update_parent_job_spec.rb:13


#### Lib - 6

- [ ] 	5 - uninitialized constant MetsDirectoryScanner::Fcntl - /spec/lib/mets_directory_scanner_spec.rb:48
- [ ] 	1 - good job count not changed - /spec/lib/activity_stream_reader_spec.rb:285

#### Mailers - 0

#### Models - 35

#### Api/ - 0 

#### Concerns/ - 4
- [ ]  		3 - ERROR:  syntax error at or near "=>" - /spec/models/concerns/delayable_spec.rb:15
- [ ] 		1 - stalled - spec/models/concerns/json_file_spec.rb

#### Open with Permission/ - 0

#### Preservica/ - 16
- [ ] 		5 - stalled - spec/models/preservica/preservica_child_error_spec.rb
- [ ] 		1 - stalled - spec/models/preservica/preservica_import_spec.rb
- [ ] 		1 - stalled - spec/models/preservica/preservica_import_pattern_two_spec.rb
- [ ] 		3 - Dif in ingest events - /spec/models/preservica/preservica_parent_error_spec.rb
- [ ] 		1 - stalled - spec/models/preservica/preservica_sequential_add_sync_spec.rb
- [ ] 		1 - stalled - spec/models/preservica/preservica_sequential_generation_bitstream_sync_spec.rb
- [ ] 		1 - stalled - spec/models/preservica/preservica_sequential_subtract_sync_spec.rb
- [ ] 		1 - stalled - spec/models/preservica/preservica_sequential_sync_spec.rb
- [ ] 		1 - stalled - spec/models/preservica/preservica_sync_spec.rb
- [ ] 		1 - stalled - spec/models/preservica/preservica_update_spec.rb

#### Models cont/ - 15
- [ ] 		1 - stalled - spec/models/batch_connection_spec.rb
- [ ] 		1 - stalled - spec/models/batch_process_create_parent_spec.rb
- [ ] 		1 - stalled - spec/models/batch_process_reassociation_redirect_spec.rb
- [ ] 		1 - stalled - spec/models/batch_process_reassociation_spec.rb
- [ ] 		1 - stalled - spec/models/batch_process_spec.rb
- [ ] 		1 - stalled - spec/models/batch_process_update_child_objects_spec.rb
- [ ] 		1 - stalled - spec/models/csv_export_spec.rb
- [ ] 		1 - stalled - spec/models/iiif_presentation_spec_v3.rb
- [ ] 		3 - Failure/Error: DatabaseCleaner.clean cause it’s set to prod env - spec/models/mets_document_spec.rb
- [ ] 		1 - stalled - spec/models/parent_object_solr_spec.rb
- [ ] 		1 - stalled - spec/models/parent_object_spec.rb
- [ ] 		1 - stalled - spec/models/parent_object_statable_spec.rb
- [ ] 		1 - expected kwargs got hash - spec/models/pyramidal_tiff_spec.rb:199

#### Requests - 30
- [ ] 1 - ERROR:  syntax error at or near "=>" - spec/requests/batch_processes_request_spec.rb:141
- [ ] 16 - it’s all outdated - spec/requests/delayed_job_dashboard_request_spec.rb
- [ ] 2 - diff in status codes - spec/requests/download_original_spec.rb
- [ ] 7 - diff in expected result - spec/requests/oid_minter_request_spec.rb
- [ ] 4 - ERROR:  syntax error at or near "=>" - /spec/requests/parent_objects_spec.rb

#### Routing - 0

#### Services - 16
- [ ] 16 - duplicate key value metadata sources - spec/services/s3_service_spec.rb

#### Views - 1
- [ ] 1 - ActionView::MissingTemplate - /spec/views/users/index.html.erb_spec.rb

**Comment by K8Sewell** [comment]
Time spent so far - 8 weeks
Remaining work:

- [ ] Upgrade Blacklight - moved to ticket #2631

**Comment by K8Sewell** [comment]
With most recent commit we sit at about 150 failing tests.  They fail for 3 main reasons: objects are not persisting, test not recognizing log in of user, and an unexpected alert.  The alert has an easy fix and the other two issues are still in progress to determine the cause of the failures.

<details>

- [ ]  ./spec/datatables/admin_set_datatable_spec.rb:10 # AdminSetDatatable admin set data tables can handle an empty model set
- [ ]  ./spec/datatables/admin_set_datatable_spec.rb:16 # AdminSetDatatable admin set data tables can handle a populated set
- [ ]  ./spec/helpers/json_helper_spec.rb:17 # JsonHelper formats json
- [ ]  ./spec/models/admin_set_spec.rb:23 # AdminSet user roles adds a viewer
- [ ]  ./spec/models/admin_set_spec.rb:30 # AdminSet user roles removes a viewer
- [ ]  ./spec/models/admin_set_spec.rb:37 # AdminSet user roles adds an editor
- [ ]  ./spec/models/admin_set_spec.rb:44 # AdminSet user roles removes an editor
- [ ]  ./spec/models/admin_set_spec.rb:51 # AdminSet user roles removes an editor when a viewer is added
- [ ]  ./spec/models/admin_set_spec.rb:59 # AdminSet user roles removes a viewer when an editor is added
- [ ]  ./spec/models/batch_process_create_parent_spec.rb:40 # BatchProcess with the metadata cloud mocked Create Parent Object batch process with a csv can create a parent object from aspace
- [ ]  ./spec/models/batch_process_create_parent_spec.rb:49 # BatchProcess with the metadata cloud mocked Create Parent Object batch process with a csv can create a parent_object
- [ ]  ./spec/models/batch_process_spec.rb:247 # BatchProcess running the background jobs xml file import importing a csv creates batch connections for all parent and child objects in the batch process
- [ ]  ./spec/models/batch_process_spec.rb:310 # BatchProcess running the background jobs xml file import running the background jobs creates a parent object with the expected values and child objects with expected values
- [ ]  ./spec/models/batch_process_spec.rb:373 # BatchProcess running the background jobs with the metadata cloud mocked creating a ParentObject from an import can create a parent_object from an array of oids
- [ ]  ./spec/models/batch_process_spec.rb:383 # BatchProcess running the background jobs with the metadata cloud mocked creating a ParentObject from a Preservica csv can create a parent_object with preservica values
- [ ]  ./spec/models/batch_process_spec.rb:404 # BatchProcess running the background jobs with the metadata cloud mocked creating a ParentObject from a Preservica csv does not create a parent_object with invalid preservica csv values
- [ ]  ./spec/models/batch_process_spec.rb:426 # BatchProcess running the background jobs with the metadata cloud mocked deleting a ParentObject from an import can delete a parent_object from an array of oids
- [ ]  ./spec/models/batch_process_spec.rb:450 # BatchProcess running the background jobs with the metadata cloud mocked deleting a ChildObject from an import can delete a parent_object from an array of oids
- [ ]  ./spec/models/batch_process_spec.rb:489 # BatchProcess running the background jobs with the metadata cloud mocked csv file import can refresh the ParentObjects from the MetadataCloud
- [ ]  ./spec/models/batch_process_spec.rb:51 # BatchProcess not running the background jobs creates a parent object with values only from the METs document
- [ ]  ./spec/models/batch_process_spec.rb:544 # BatchProcess running the background jobs with the metadata cloud mocked csv file import with a parent object that had been previously created and user with editor role does not alter already-existing parent object
- [ ]  ./spec/models/batch_process_spec.rb:591 # BatchProcess running the background jobs with the metadata cloud mocked uploading a csv of oids to create parent objects succeeds if the user is an editor on the admin set of the parent object
- [ ]  ./spec/models/batch_process_spec.rb:598 # BatchProcess running the background jobs with the metadata cloud mocked uploading a csv of oids to create parent objects fails if the user is not an editor on the admin set of the parent object
- [ ]  ./spec/models/concerns/reassociatable_spec.rb:11 # Reassociatable blanks out fields which can be blanked
- [ ]  ./spec/models/concerns/reassociatable_spec.rb:21 # Reassociatable does not blank out order, which can not be blanked
- [ ]  ./spec/models/open_with_permission/permission_set_spec.rb:12 # OpenWithPermission::PermissionSet user permission set roles adds an approver
- [ ]  ./spec/models/open_with_permission/permission_set_spec.rb:19 # OpenWithPermission::PermissionSet user permission set roles removes an approver
- [ ]  ./spec/models/open_with_permission/permission_set_spec.rb:26 # OpenWithPermission::PermissionSet user permission set roles adds an administrator
- [ ]  ./spec/models/open_with_permission/permission_set_spec.rb:33 # OpenWithPermission::PermissionSet user permission set roles removes an administrator
- [ ]  ./spec/models/open_with_permission/permission_set_spec.rb:40 # OpenWithPermission::PermissionSet user permission set roles removes an administrator when a approver is added
- [ ]  ./spec/models/open_with_permission/permission_set_spec.rb:48 # OpenWithPermission::PermissionSet user permission set roles removes an approver when an administrator is added
- [ ]  ./spec/models/parent_object_spec.rb:285 # ParentObject a newly created ParentObject with different visibilities visibility defaults to Private if not a valid visibility
- [ ]  ./spec/models/parent_object_spec.rb:297 # ParentObject a newly created ParentObject with different visibilities open with Permission visibility does not validate without a permission set
- [ ]  ./spec/models/preservica/preservica_child_error_spec.rb:97 # Preservica::PreservicaObject can create child objects
- [ ]  ./spec/requests/admin_sets_spec.rb:108 # /admin_sets when user has permission to Sets PATCH /update with valid parameters updates the requested admin_set
- [ ]  ./spec/requests/admin_sets_spec.rb:115 # /admin_sets when user has permission to Sets PATCH /update with valid parameters redirects to the admin_set
- [ ]  ./spec/requests/admin_sets_spec.rb:124 # /admin_sets when user has permission to Sets PATCH /update with invalid parameters renders a successful response (i.e. to display the 'edit' template)
- [ ]  ./spec/requests/admin_sets_spec.rb:133 # /admin_sets when user has permission to Sets DELETE /destroy destroys the requested admin_set
- [ ]  ./spec/requests/admin_sets_spec.rb:140 # /admin_sets when user has permission to Sets DELETE /destroy redirects to the admin_sets list
- [ ]  ./spec/requests/admin_sets_spec.rb:153 # /admin_sets when user does not permission to Sets GET /index renders an unauthorized message
- [ ]  ./spec/requests/admin_sets_spec.rb:161 # /admin_sets when user does not permission to Sets GET /show renders an unauthorized message
- [ ]  ./spec/requests/admin_sets_spec.rb:44 # /admin_sets when user has permission to Sets GET /index renders a successful response
- [ ]  ./spec/requests/admin_sets_spec.rb:52 # /admin_sets when user has permission to Sets GET /show renders a successful response
- [ ]  ./spec/requests/admin_sets_spec.rb:60 # /admin_sets when user has permission to Sets GET /new renders a successful response
- [ ]  ./spec/requests/admin_sets_spec.rb:67 # /admin_sets when user has permission to Sets GET /edit render a successful response
- [ ]  ./spec/requests/admin_sets_spec.rb:76 # /admin_sets when user has permission to Sets POST /create with valid parameters creates a new AdminSet
- [ ]  ./spec/requests/admin_sets_spec.rb:82 # /admin_sets when user has permission to Sets POST /create with valid parameters redirects to the created admin_set
- [ ]  ./spec/requests/admin_sets_spec.rb:89 # /admin_sets when user has permission to Sets POST /create with invalid parameters does not create a new AdminSet
- [ ]  ./spec/requests/admin_sets_spec.rb:95 # /admin_sets when user has permission to Sets POST /create with invalid parameters renders a successful response (i.e. to display the 'new' template)
- [ ]  ./spec/requests/download_original_spec.rb:44 # Download Original API POST /api/download/stage/child/:oid creates a new job to copy to s3
- [ ]  ./spec/requests/download_original_spec.rb:50 # Download Original API POST /api/download/stage/child/:oid errors if object is not YCO or Public
- [ ]  ./spec/requests/download_original_spec.rb:56 # Download Original API POST /api/download/stage/child/:oid errors if child oid is not found
- [ ]  ./spec/system/admin_set_spec.rb:113 # Admin Sets when user has permission to Sets display admin edit form
- [ ]  ./spec/system/admin_set_spec.rb:123 # Admin Sets when user has permission to Sets allow editing using form
- [ ]  ./spec/system/admin_set_spec.rb:134 # Admin Sets when user has permission to Sets allow new using form
- [ ]  ./spec/system/admin_set_spec.rb:147 # Admin Sets when user has permission to Sets create fails with invalid url
- [ ]  ./spec/system/admin_set_spec.rb:159 # Admin Sets when user has permission to Sets create does not submit with missing label
- [ ]  ./spec/system/admin_set_spec.rb:169 # Admin Sets when user has permission to Sets edit fails with invalid url
- [ ]  ./spec/system/admin_set_spec.rb:176 # Admin Sets when user has permission to Sets edit does not submit with missing label
- [ ]  ./spec/system/admin_set_spec.rb:18 # Admin Sets when user has permission to Sets display admin sets
- [ ]  ./spec/system/admin_set_spec.rb:184 # Admin Sets when user has permission to Sets the label appears on the slide bar
- [ ]  ./spec/system/admin_set_spec.rb:189 # Admin Sets when user has permission to Sets has update metadata button
- [ ]  ./spec/system/admin_set_spec.rb:202 # Admin Sets when user has permission to Sets update metadata dialog autoselects the current admin set
- [ ]  ./spec/system/admin_set_spec.rb:206 # Admin Sets when user has permission to Sets update metadata dialog requires selection of metadata sources
- [ ]  ./spec/system/admin_set_spec.rb:215 # Admin Sets when user has permission to Sets update metadata dialog starts job when dialog is submitted
- [ ]  ./spec/system/admin_set_spec.rb:23 # Admin Sets when user has permission to Sets displays the user roles tables
- [ ]  ./spec/system/admin_set_spec.rb:244 # Admin Sets when user does not have permission to Sets cannot update iiif manifests without edit permission
- [ ]  ./spec/system/admin_set_spec.rb:29 # Admin Sets when user has permission to Sets validates preservica credentials
- [ ]  ./spec/system/admin_set_spec.rb:34 # Admin Sets when user has permission to Sets allows roles to be added to users
- [ ]  ./spec/system/admin_set_spec.rb:50 # Admin Sets when user has permission to Sets allows roles to be removed from users
- [ ]  ./spec/system/admin_set_spec.rb:66 # Admin Sets when user has permission to Sets can export parent objects
- [ ]  ./spec/system/admin_set_spec.rb:76 # Admin Sets when user has permission to Sets can update iiif manifests as an admin set editor
- [ ]  ./spec/system/admin_set_spec.rb:85 # Admin Sets when user has permission to Sets cannot update iiif manifests if not an editor of the admin set
- [ ]  ./spec/system/admin_set_spec.rb:93 # Admin Sets when user has permission to Sets removes the viewer role from a user when they are given an editor role
- [ ]  ./spec/system/batch_process_child_detail_spec.rb:39 # Batch Process Child detail page with expected success with a csv import has a link to the batch process detail page
- [ ]  ./spec/system/batch_process_detail_spec.rb:107 # Batch Process detail page when uploading a csv deleting a parent object can still see the details of the import
- [ ]  ./spec/system/batch_process_detail_spec.rb:126 # Batch Process detail page when uploading an xml doc can see the details of the import
- [ ]  ./spec/system/batch_process_detail_spec.rb:146 # Batch Process detail page when uploading an xml doc without intranda information can see the details of the import
- [ ]  ./spec/system/batch_process_detail_spec.rb:46 # Batch Process detail page when uploading a csv can see the details of the import
- [ ]  ./spec/system/batch_process_detail_spec.rb:71 # Batch Process detail page when uploading a csv when batch wide ingest event is available can see the ingest events of the import
- [ ]  ./spec/system/batch_process_detail_spec.rb:77 # Batch Process detail page when uploading a csv can see the status of the parent object imports
- [ ]  ./spec/system/batch_process_detail_spec.rb:82 # Batch Process detail page when uploading a csv can see the overall status of the batch process
- [ ]  ./spec/system/batch_process_detail_spec.rb:88 # Batch Process detail page when uploading a csv can see the overall status of the batch process BAD admin set
- [ ]  ./spec/system/batch_process_parent_detail_spec.rb:64 # Batch Process Parent detail page with expected success with a csv import running the background jobs has a child object id
- [ ]  ./spec/system/batch_process_parent_detail_spec.rb:68 # Batch Process Parent detail page with expected success with a csv import has a link to the batch process detail page
- [ ]  ./spec/system/batch_process_parent_detail_spec.rb:72 # Batch Process Parent detail page with expected success with a csv import has a link to the parent object page
- [ ]  ./spec/system/batch_process_parent_detail_spec.rb:76 # Batch Process Parent detail page with expected success with a csv import shows the status of the parent object
- [ ]  ./spec/system/batch_process_parent_detail_spec.rb:80 # Batch Process Parent detail page with expected success with a csv import shows when the parent object was submitted
- [ ]  ./spec/system/batch_process_parent_detail_spec.rb:86 # Batch Process Parent detail page with expected success with a csv import has labels for the ingest steps for the parent object
- [ ]  ./spec/system/batch_process_preservica_spec.rb:70 # BatchProcess user with edit permission can recognize and report a failure
- [ ]  ./spec/system/batch_process_reassociation_redirect_spec.rb:43 # BatchProcess child object reassociation with removing all children updates relationships and creates a redirected parent object
- [ ]  ./spec/system/batch_process_reassociation_redirect_spec.rb:65 # BatchProcess child object reassociation with redirected parent object destination does not update relationships and does not change the redirected parent object
- [ ]  ./spec/system/batch_process_reassociation_spec.rb:100 # BatchProcess child object reassociation with an invalid viewing hint returns error message when invalid controlled vocabularly is present in csv
- [ ]  ./spec/system/batch_process_reassociation_spec.rb:38 # BatchProcess child object reassociation with missing columns does not update already existing values if column is missing
- [ ]  ./spec/system/batch_process_reassociation_spec.rb:76 # BatchProcess child object reassociation with all columns present updates value to nil when nil value present in csv
- [ ]  ./spec/system/batch_process_spec.rb:105 # BatchProcess when uploading a csv re-associating child objects displays children in batch parent details
- [ ]  ./spec/system/batch_process_spec.rb:120 # BatchProcess when uploading a csv re-associating child objects displays batch messages on batch show
- [ ]  ./spec/system/batch_process_spec.rb:130 # BatchProcess when uploading a csv re-associating child objects displays batch messages for invalid order
- [ ]  ./spec/system/batch_process_spec.rb:143 # BatchProcess when uploading a csv re-associating child objects updates all parent object counts
- [ ]  ./spec/system/batch_process_spec.rb:152 # BatchProcess when uploading a csv re-associating child objects does not update label, caption, or order if not in csv
- [ ]  ./spec/system/batch_process_spec.rb:191 # BatchProcess when uploading a csv outputting csv uploads a CSV of parent objects in order to create export of parent objects metadata
- [ ]  ./spec/system/batch_process_spec.rb:205 # BatchProcess when uploading a csv outputting csv uploads a CSV of parent objects in order to create export of parent objects metadata fails 2005512 row due to admin set permissions
- [ ]  ./spec/system/batch_process_spec.rb:218 # BatchProcess when uploading a csv outputting csv uploads a CSV of admin set in order to create export of parent object oids
- [ ]  ./spec/system/batch_process_spec.rb:236 # BatchProcess when uploading a csv outputting csv uploads a CSV of parent oids in order to create export of child objects oids and orders
- [ ]  ./spec/system/batch_process_spec.rb:276 # BatchProcess when uploading a csv outputting csv round-tripping csv can create the output csv from a csv that has been generated from the application
- [ ]  ./spec/system/batch_process_spec.rb:286 # BatchProcess when uploading a csv outputting csv round-tripping csv can create the output csv from a handmade csv
- [ ]  ./spec/system/batch_process_spec.rb:312 # BatchProcess when uploading a csv deleting a parent object can still see the details of the import
- [ ]  ./spec/system/batch_process_spec.rb:331 # BatchProcess when uploading a csv re-generate ptiffs child objects displays children in batch parent details
- [ ]  ./spec/system/batch_process_spec.rb:34 # BatchProcess when uploading a csv uploads and increases csv count and gives a success message
- [ ]  ./spec/system/batch_process_spec.rb:343 # BatchProcess when uploading a csv re-generate ptiffs child objects displays batch messages on batch details
- [ ]  ./spec/system/batch_process_spec.rb:355 # BatchProcess when uploading an xml uploads and increases xml count and gives a success message
- [ ]  ./spec/system/batch_process_spec.rb:371 # BatchProcess when uploading an xml deleting a parent object can still load the batch_process page
- [ ]  ./spec/system/batch_process_spec.rb:396 # BatchProcess when uploading an xml with jobs running create preservica ingest for the parent and children objects
- [ ]  ./spec/system/batch_process_spec.rb:426 # BatchProcess triggers directory scan
- [ ]  ./spec/system/batch_process_spec.rb:56 # BatchProcess when uploading a csv errors batch if CSV contains too many entries
- [ ]  ./spec/system/batch_process_spec.rb:67 # BatchProcess when uploading a csv errors batch if CSV does not contain any row data
- [ ]  ./spec/system/batch_process_spec.rb:96 # BatchProcess when uploading a csv re-associating child objects uploads a CSV of child oids in order to re-associate them with new parent oids
- [ ]  ./spec/system/batch_process_update_parent_spec.rb:116 # BatchProcess with a user with edit permissions but with only an oid value updating a batch of parent objects triggers a metadata update
- [ ]  ./spec/system/batch_process_update_parent_spec.rb:147 # BatchProcess with a user with edit permissions but with invalid metadata source value updating a batch of parent objects triggers a metadata update
- [ ]  ./spec/system/batch_process_update_parent_spec.rb:170 # BatchProcess with a user with edit permissions but with a redirect with correct format updating a batch of parent objects does trigger a metadata update
- [ ]  ./spec/system/batch_process_update_parent_spec.rb:199 # BatchProcess with a user with edit permissions but with a redirect with incorrect format updating a batch of parent objects does not trigger a metadata update
- [ ]  ./spec/system/batch_process_update_parent_spec.rb:221 # BatchProcess with a user with edit permissions but with a redirect when child objects are present updating a batch of parent objects does not trigger a metadata update
- [ ]  ./spec/system/batch_process_update_parent_spec.rb:247 # BatchProcess with a user without edit permissions updating a batch of parent objects does not permit parent to be updated
- [ ]  ./spec/system/batch_process_update_parent_spec.rb:33 # BatchProcess with a user with edit permissions and valid controlled vocabulary updating a batch of parent objects updates the parent
- [ ]  ./spec/system/batch_process_update_parent_spec.rb:70 # BatchProcess with a user with edit permissions but without valid controlled vocabulary updating a batch of parent objects does not update the parent with invalid values
- [ ]  ./spec/system/child_object_spec.rb:63 # ChildObjects when logged in with access to only some admin set roles does not allow viewing of the child object the user does not have access to
- [ ]  ./spec/system/parent_object_list_spec.rb:20 # ParentObjects parent objects datatable page displays parent objects without filter
- [ ]  ./spec/system/parent_object_list_spec.rb:27 # ParentObjects parent objects datatable page filters when filter box is filled
- [ ]  ./spec/system/parent_object_list_spec.rb:35 # ParentObjects parent objects datatable page retains filters and fills filter boxes on refresh
- [ ]  ./spec/system/parent_object_list_spec.rb:48 # ParentObjects parent objects datatable page clears filter when Clear Filters is clicked
- [ ]  ./spec/system/parent_object_spec.rb:455 # ParentObjects editing a ParentObject validates redirect presence
- [ ]  ./spec/system/parent_object_spec.rb:530 # ParentObjects index page datatable has multiple Parent Objects
- [ ]  ./spec/system/parent_object_spec.rb:552 # ParentObjects index page index page datatable has multiple Parent Objects
- [ ]  ./spec/system/parent_object_spec.rb:566 # ParentObjects index page index page clicking ReIndex button does not Reindex if a reindex job is already in progress
- [ ]  ./spec/system/parent_object_spec.rb:573 # ParentObjects index page index page clicking ReIndex button does not Reindex without confirmation
- [ ]  ./spec/system/parent_object_spec.rb:579 # ParentObjects index page index page clicking ReIndex button does Reindex with confirmation
- [ ]  ./spec/system/parent_object_spec.rb:615 # ParentObjects index page index page logged in with editor rights does allow create parent
- [ ]  ./spec/system/parent_object_spec.rb:619 # ParentObjects index page index page logged in with editor rights does not allow editing of oid for non-sysadmin
- [ ]  ./spec/system/parent_object_spec.rb:635 # ParentObjects index page when logged in without admin set roles does not allow creation of new parent with wrong admin set
- [ ]  ./spec/system/parent_object_spec.rb:661 # ParentObjects index page when logged in with access to only some admin set roles does not display parent objects the user does not have access to view
- [ ]  ./spec/system/parent_object_spec.rb:674 # ParentObjects index page when logged in with access to only some admin set roles does not allow viewing of the parent object the user does not has access to
- [ ]  ./spec/system/parent_object_spec.rb:684 # ParentObjects index page when logged in with access to only some admin set roles does not allow viewing of the child object the user does not has access to
- [ ]  ./spec/system/parent_object_spec.rb:689 # ParentObjects index page when logged in with access to only some admin set roles does not allow changing parent_object to admin_set user does not have access to
- [ ]  ./spec/system/parent_object_spec.rb:724 # ParentObjects parent objects page has column visibility button
- [ ]  ./spec/system/structure_editor_spec.rb:104 # Structure Editor can perform actions and can submit structure back to management
- [ ]  ./spec/system/structure_editor_spec.rb:46 # Structure Editor can access the homepage can render a manifest
- [ ]  ./spec/system/structure_editor_spec.rb:64 # Structure Editor can perform actions and can add a range
- [ ]  ./spec/system/structure_editor_spec.rb:69 # Structure Editor can perform actions and can add a canvas
- [ ]  ./spec/system/structure_editor_spec.rb:77 # Structure Editor can perform actions and can delete a range
- [ ]  ./spec/system/structure_editor_spec.rb:84 # Structure Editor can perform actions and can delete a canvas
- [ ]  ./spec/system/structure_editor_spec.rb:95 # Structure Editor can perform actions and can change range title
- [ ]  ./spec/system/user_spec.rb:111 # Users are editable and can be deactivated
- [ ]  ./spec/system/user_spec.rb:121 # Users are editable and can be made sysadmin
- [ ]  ./spec/system/user_spec.rb:132 # Users created with valid user succeeds
- [ ]  ./spec/system/user_spec.rb:146 # Users with invalid properties provides feedback
- [ ]  ./spec/system/user_spec.rb:21 # Users datatable defaults to only display Active users
- [ ]  ./spec/system/user_spec.rb:27 # Users datatable defaults to ascending order
- [ ]  ./spec/system/user_spec.rb:39 # Users datatable filters by netid
- [ ]  ./spec/system/user_spec.rb:65 # Users are viewable displays the users roles
- [ ]  ./spec/system/user_spec.rb:76 # Users are editable and require an email to be present
- [ ]  ./spec/system/user_spec.rb:89 # Users are editable and require their full name to be present
- [ ]  ./spec/views/problem_reports/index.html.erb_spec.rb:6 # problem_reports/index renders a list of problem_reports
- [ ]  ./spec/views/users/index.html.erb_spec.rb:6 # users/index.html.erb displays the users datatable
</details>

**Comment by K8Sewell** [comment]
Tickets created to fix remaining broken specs: 

- [ ] #2699
- [ ] #2700
- [ ] #2701

**Comment by K8Sewell** [comment]
PRs ready for review:

Camerata - https://github.com/yalelibrary/yul-dc-camerata/pull/365

Management - https://github.com/yalelibrary/yul-dc-management/pull/1299

Blacklight - in another ticket

**Comment by K8Sewell** [comment]
Waiting on test plan

**Comment by K8Sewell** [comment]
Deployed to Test and Demo with release [v2.67.1](https://github.com/yalelibrary/yul-dc-management/releases/tag/v2.67.1)

**Comment by K8Sewell** [comment]
Test environment is live with Management v2.67.2 so this is ready to be tested.

**Comment by sshetenhelm** [comment]
There may be some issues between Preservica and DCS that need to be further investigated. Will do more testing and report back.

Other aspects of Management look good.

**Comment by sshetenhelm** [comment]
Preservica upload is not working in UAT. Unsure if it is due to this bug fix or the Ruby upgrade. 

When attempting to upload, I received the following error: 
`Child Object 901622792 failed to convert PTIFF due to Expected file originals/92/90/16/22/79/901622792.tif not found.`
So I guess the child image did not successfully copy over from Preservica? 
[Video](https://drive.google.com/file/d/1RIYJOnHKAu-Do3EMgZk8pSKYvFTNHC2e/view?usp=sharing)

Will also comment in Preservica bug ticket.

**Comment by sshetenhelm** [comment]
Great news! Got an object to go through UAT -- not the checksum test one, but a different one, so likely the checksum one is still causing errors but maybe the non-checksum errors are fixed? 
Batch process -- https://collections-uat.library.yale.edu/management/batch_processes/1881/parent_objects/901623825

I will test again before we officially celebrate.

**Comment by sshetenhelm** [comment]
...but these two failed: 
https://collections-uat.library.yale.edu/management/batch_processes/1882

I don't know why I received the image error for these two but not for the above.
